[IMP] l10n_mx: Add 19.2 changes to the MX loc (#17376)
odoo/documentation: content/applications/finance/fiscal_localizations/mexico.rst

   On a testing environment, the message :guilabel:`Not Found` will always come up.

.. warning::

>    Invoices with a :guilabel:`Country` different to Mexico or missing the :guilabel:`Zip` field in

Reviewer: ```suggestion
   Invoices with a :guilabel:`Country` different from Mexico or missing the :guilabel:`Zip` field in
```

.. warning::
   If the partner does not contain the :guilabel:`Country` or :guilabel:`Zip` the invoicing is
   assumed as a |CFDI| to public and the invoicing will be addressed to the generic customer, the
   checkbox will remain active as long as the partner stays invalid, if the invoice is to be adressed
   to an actual national customer please complete the contact information and the checkbox will be

>    unticked automatically.

Reviewer: I wasn't able to test it, so feel free to adapt if there's anything that needs to be changed.
```suggestion
Invoices with incomplete :guilabel:`Country` or :guilabel:`Zip` fields in the customer form are
classified as :guilabel:`CFDI to public` and addressed to a generic customer. The :guilabel:`CFDI to
public` checkbox stays active on the invoice form until valid partner information is provided. Once
the contact information for a national customer is complete, the checkbox is automatically
deactivated.
```

Because of this requirement, the field :guilabel:`CFDI Origin` adds this relation with a `01|`,
followed by the fiscal folio of the original :abbr:`UUID (universally unique identifier)`.

While creating credit notes the description of the lines needs to indicate what it is for, to do
this just modify the line descrpition and explain in text what is being given as a bonus, discount,

> or refund.

Reviewer: I tried to simplify this paragraph and make it more direct.
```suggestion
When creating credit notes, make sure the line description clearly indicates its purpose and
specifies which entry applies to a bonus, discount, or refund.
```

   .. tip::
      - The :guilabel:`Amount` and :guilabel:`Compensation` amounts are provided by the factor and
        are stated on the factoring receipt.
      - Make sure the :guilabel:`Amount` and :guilabel:`Compensation` amounts equal the
        :guilabel:`Due` amount. If it exceeds the :guilabel:`Due` amount, the line is highlighted in
        red.
      - The :guilabel:`To distribute` field should balance the :guilabel:`Amount` to the transaction
        amount entered, ensuring it equals zero.

#. Click :guilabel:`Save`

In the :guilabel:`Bank Matching` view, the factoring payment is then reconciled with the relevant
invoices, and the factoring commission is allocated to the :guilabel:`Factoring` account set up.

To generate the XML file , go to :menuselection:`Accounting --> Customers --> Invoices`, access one
of the paid invoices and click on :guilabel:Update Payments."

.. note::
   - The factoring company can retain a portion of the invoice amount, referred to as aforo, until
   the end customer completes payment. The retained balance is then released to the vendor.
   - The hidden field :guilabel:`Due (Company)` keeps the amount in company currency when dealing

>    with multi-currency factoring.

Reviewer: ```suggestion
   - For multi-currency factoring, use the :icon:`oi-settings-adjust` (:guilabel:`adjust settings`)
     button in the :guilabel:`Distribute Payment` window and add the :guilabel:`Due (Company)`
     column to display the due amount in the company's currency.
```

Reviewer: Thank you for the screenshot! I tried to be as complete as possible, feel free to update if needed ;)

.. image:: mexico/mx-print-payment.png
   :alt: Example of the print button on the CFDI tab.

.. _l10n/mx/factoring:

Factoring
*********

Factoring is a financial transaction in which a business sells its accounts receivable (outstanding

> invoices) to a specialized financial institution, known as a "factor", to gain immediate liquidity.

Reviewer 2: ```suggestion
invoices) to a specialized financial institution, known as a *factor*, to gain immediate liquidity.
```

To record a payment for financial factoring, follow these steps:

#. Go to :menuselection:`Accounting --> Configuration --> Settings`, scroll down to the
   :guilabel:`Default Accounts` section to make sure a :guilabel:`Factoring` account is set up in

>    the :guilabel:`Factoring account` section.

Reviewer 2: ```suggestion
#. Go to :menuselection:`Accounting --> Configuration --> Settings`, scroll down to the
   :guilabel:`Default Accounts` section and ensure a :guilabel:`Factoring account`  is selected in
   the :guilabel:`Factoring account` field.
```
Both words are visible on the UI, so even though it's odd to see both bold in a row, it needs to match.

> #. From the Accounting Dashboard, click the :guilabel:`Bank` journal’s name to manually :ref:`add a

Reviewer 2: ```suggestion
#. From the Accounting Dashboard, click the word :guilabel:`Bank` on it's corresponding Kanban card to manually :ref:`add a
```
I think this is a bit clearer - I was looking for the journal name I set in the section above at first.

   transaction line <accounting/transactions/register>` and record the factoring payment. Complete
   the following fields:

>    - :guilabel:`Label`

Reviewer 2: ```suggestion
   - :guilabel:`Label`: Enter a brief description of the transaction.
```

This should not be blank - so please adjust if this is incorrect, I just guessed!

   - :guilabel:`Partner`: Select the financial institution (factor) that will purchase the debt.
   - :guilabel:`Amount`: Enter the actual amount to be received by the factor.

   .. important::

>       The currency of the factoring payment must match the bank journal's currency.

Reviewer 2: ```suggestion
      The currency of the factoring payment **must** match the bank journal's currency.
```
just for emphasis

#. In the newly created transaction line, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical
   ellipsis)` icon and select :guilabel:`Reconcile`.
#. In the :guilabel:`Search: Journal Items to Match` window, remove the :guilabel:`Partner` filter,
   select the related invoice(s) to be paid by the factor, and click :guilabel:`Select &

>    Distribute`.

Reviewer 2: ```suggestion
   select the related invoices to be paid by the factor, and click :guilabel:`Select`.
```
When this occurs, we always default to the plural. Also, the button I saw was just 'select' - perhaps I did it wrong, so please double check.

   .. note::

>       Do not select the tax lines related to the invoices.

Reviewer 2: I woudl mention why - does this cause an error? Is this not legal? Etc

Author: Its not related to the MX loc, its just that Odoo shows them up for some reason even though they are useless here, its a helpful note.

#. In the :guilabel:`Distribute Payment` window, fill in the following fields to distribute the

>    payment on the selected invoice(s):

Reviewer 2: ```suggestion
   payment on the selected invoices:
```

   - :guilabel:`Amount`: Enter the amount that will actually be received from the factor.
   - :guilabel:`Compensation`: Enter the compensation that was paid to the factor.

   .. tip::
      - The :guilabel:`Amount` and :guilabel:`Compensation` amounts are provided by the factor and
        are stated on the factoring receipt.
      - Make sure the :guilabel:`Amount` and :guilabel:`Compensation` amounts equal the
        :guilabel:`Due` amount. If it exceeds the :guilabel:`Due` amount, the line is highlighted in
        red.
      - The :guilabel:`To distribute` field should balance the :guilabel:`Amount` to the transaction
        amount entered, ensuring it equals zero.

#. Click :guilabel:`Save`

In the :guilabel:`Bank Matching` view, the factoring payment is then reconciled with the relevant
invoices, and the factoring commission is allocated to the :guilabel:`Factoring` account set up.

To generate the XML file , go to :menuselection:`Accounting --> Customers --> Invoices`, access one

> of the paid invoices and click on :guilabel:Update Payments."

Reviewer: Some typos and to fix the Documentation guidelines check issue.
```suggestion
To generate the XML file, go to :menuselection:`Accounting --> Customers --> Invoices`, access one
of the paid invoices, and click :guilabel:`Update Payments`.
```

.. note::

>    - The factoring company can retain a portion of the invoice amount, referred to as aforo, until
>    the end customer completes payment. The retained balance is then released to the vendor.
>    - For multi-currency factoring, use the :icon:`oi-settings-adjust` (:guilabel:`adjust settings`)
>      button in the :guilabel:`Distribute Payment` window and add the :guilabel:`Due (Company)`
>      column to display the due amount in the company's currency.

Reviewer: There was just an issue with the alignment here, so the documentation check didn't pass. 

```suggestion
   - The factoring company can retain a portion of the invoice amount, referred to as aforo, until
     the end customer completes payment. The retained balance is then released to the vendor.
   - For multi-currency factoring, use the :icon:`oi-settings-adjust` (:guilabel:`adjust settings`)
     button in the :guilabel:`Distribute Payment` window and add the :guilabel:`Due (Company)`
     column to display the due amount in the company's currency.
```

.. _l10n/mx/factoring:

Factoring
*********

To create a payment for financial factoring, create a payment directly from a bank journal by
:doc:`adding a statement line <../accounting/payments>`, when adding a new payment instead of 
setting the customer of the invoice as contact, set the financial insitution (factorant) that will
purchase the debt and input the actual amount of money to be received by the factorant, after that
select :guilabel:`Reconcile` and the regular reconciliation menu will open, here remove the partner
filter and select the invoices that will be paid by the factorant and click on :guilabel:`Select &
Distribute`

.. image:: mexico/mx-factoring-distribution.png
   :alt: Example of a distribution of 9000MXN to two invoices adding up to 10440MXN.

On the distribution menu each line represents an invoice that will recieve a payment, the amount of
the payment *must* be distributed on the selected invoices by filling the :guilabel:`Amount` field
and register the compensation per line on the :guilabel:`Compensation` field and click on
:guilabel:`Save`.

Additionally the :guilabel:`To distribute` section at the botton keeps track of the amount pending

> and will turn the last line to red if the distributed amound exceeds the payment amount.

Reviewer: I thought it would be useful for the user to add some context.
Also, I made some suggestions to make the instructions as precise as possible.
With the updated instructions, I think the image is no longer needed, as the text is clear and self-explanatory without it.
An additional information might be needed for the Label field is step 2.
```suggestion
Factoring is a financial transaction in which a business sells its accounts receivable (outstanding
invoices) to a specialized financial institution, known as a "factor", to gain immediate liquidity.

To record a payment for financial factoring, follow these steps:

#. Go to :menuselection:`Accounting --> Configuration --> Settings`, scroll down to the
   :guilabel:`Default Accounts` section to make sure a :guilabel:`Factoring` account is set up in
   the :guilabel:`Factoring account` section.
#. From the Accounting Dashboard, click the :guilabel:`Bank` journal’s name to manually :ref:`add a
   transaction line <accounting/transactions/register>` and record the factoring payment. Complete
   the following fields:

   - :guilabel:`Label`
   - :guilabel:`Partner`: Select the financial institution (factor) that will purchase the debt.
   - :guilabel:`Amount`: Enter the actual amount to be received by the factor.

   .. important::
      The currency of the factoring payment must match the bank journal's currency.

#. In the newly created transaction line, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical
   ellipsis)` icon and select :guilabel:`Reconcile`.
#. In the :guilabel:`Search: Journal Items to Match` window, remove the :guilabel:`Partner` filter,
   select the related invoice(s) to be paid by the factor, and click :guilabel:`Select &
   Distribute`.

   .. note::
      Do not select the tax lines related to the invoices.

#. In the :guilabel:`Distribute Payment` window, fill in the following fields to distribute the
   payment on the selected invoice(s):

   - :guilabel:`Amount`: Enter the amount that will actually be received from the factor.
   - :guilabel:`Compensation`: Enter the compensation that was paid to the factor.

   .. tip::
      - The :guilabel:`Amount` and :guilabel:`Compensation` amounts are provided by the factor and
        are stated on the factoring receipt.
      - Make sure the :guilabel:`Amount` and :guilabel:`Compensation` amounts equal the
        :guilabel:`Due` amount. If it exceeds the :guilabel:`Due` amount, the line is highlighted in
        red.
      - The :guilabel:`To distribute` field should balance the :guilabel:`Amount` to the transaction
        amount entered, ensuring it equals zero.

#. Click :guilabel:`Save`

In the :guilabel:`Bank Matching` view, the factoring payment is then reconciled with the relevant
invoices, and the factoring commission is allocated to the :guilabel:`Factoring` account set up.
```

Author: Beautiful

> To generate the XML file just open one of the paid invoices and click on :guilabel:`Update Payments`.

Reviewer: I wasn't able to test this part because the "Update Payments" button wasn't available in the runbot I tried after the reconciliation.
```suggestion
To generate the XML file, go to :menuselection:`Accounting --> Customers --> Invoices`, access the
paid invoice and click on :guilabel:`Update Payments`.
```

Author: What I suspect happened is that it is implicit here that the invoices must be signed first, perhaps that's why it didn't appear.

Author: I'm not sure about this one, since you can pay multiple invoices I want to clarify that any of the paid invoices can be used and not just one.
The wording here in singular implies that there was only one, doesn't it?

Reviewer: Yes, you're right, I used a singular. Maybe this:
"access one of the paid invoices and click on :guilabel:`Update Payments`."

What do you think?

Author: Pristine

.. note::
   - It is possible to leave unpaid balance for financial aforum to invoice at a later date.
   - The hidden field :guilabel:`Due (Company)` keeps the amount in company currency when dealing

>    with multi-currency factoring.

Reviewer: For the first bullet, I'm not sure I understood what it meant, since (I think) there was a typo. But I'll tend to use a more direct tone.
For the second bullet, I couldn't find the hidden field ;) but it might be useful to explain where it is.

```suggestion
   - The factoring transaction can be left with an unpaid balance to be processed at a later date.
   - The hidden :guilabel:`Due (Company)` field keeps the amount in company currency for
     multi-currency factoring.
```

Author: Here I would like to keep the word, in spanish it is "aforo" its a set amount of money that the factor leaves unpaid until the end customer pays them, only then do they give you the rest of the money, this process is common but I don't have any better words or a shorter version to explain it, thus I just wrote aforum.
The idea is that people who are looking on how to add an aforum can find it faster.

The hidden field is on the breadcrumbs

Reviewer: Thanks a lot for the explanation for the aforo! Then, maybe it can be something like this (to add a little bit of information):
- The factoring company can retain a portion of the invoice amount, referred to as *aforo*, until 
  the end customer completes payment. The retained balance is then released to the vendor.

For the Due (Company) hidden field, can you take a screenshot so I can see exactly where it is? Thank you!

Author: I'll update, meanwhile, I leave here the picture of the "Due (in company)" field.
<img width="980" height="377" alt="image" src="https://github.com/user-attachments/assets/29fe584b-7115-44f4-ab74-7bc10c8b83aa" />

.. warning::
   - If there is no account set on the factoring section in settings the distribution button will
   not be visible.
   - It is not possible to create a factoring payment if the currency of the payment is different of

>    that of the bank journal.

Reviewer: I included this information in the instructions above to help users follow the steps.
```suggestion
```
